cetlvast/suites/unittest/test_variable_length_array_compat.cpp

```diff
@@ -1036,3 +1189,68 @@ TEST(VLATestsNonTrivialSpecific, TestMoveAssignment)
     ASSERT_NE(lhs, rhs);
     ASSERT_EQ(std::string("three"), lhs[0]);
 }
+
+struct NoDefault
+{
+    ~NoDefault() = default;
+    NoDefault()  = delete;
+    NoDefault(int value)
+        : data_{value} {};
+    NoDefault(const NoDefault&)                = default;
+    NoDefault(NoDefault&&) noexcept            = default;
+    NoDefault& operator=(const NoDefault&)     = default;
+    NoDefault& operator=(NoDefault&&) noexcept = default;
+
+    int data() const noexcept
+    {
+        return data_;
+    }
+
+private:
+    int data_;
+};
+
+TEST(VLATestsNonTrivialSpecific, TestResizeWithNoDefaultCtorData)
+{
+    std::allocator<NoDefault>                                       allocator{};
+    cetl::VariableLengthArray<NoDefault, std::allocator<NoDefault>> subject{{NoDefault{1}}, allocator};
+    ASSERT_EQ(1, subject.size());
+    subject.resize(10, NoDefault{2});
+    ASSERT_EQ(10, subject.size());
+    ASSERT_EQ(1, subject[0].data());
+    for (std::size_t i = 1; i < subject.size(); ++i)
+    {
+        ASSERT_EQ(2, subject[i].data());
+    }
+}
+
+#ifdef __cpp_exceptions
+
+struct GrenadeError : public std::runtime_error
+{
+    GrenadeError(const char* message)
+        : std::runtime_error(message)
+    {
+    }
+};
+
+struct Grenade
+{
+    Grenade(int value)
+    {
+        if (value == 2)
+        {
+            throw GrenadeError("");
+        }
+    }
+};
+
+TYPED_TEST(VLATestsGeneric, TestResizeExceptionFromCtorOnResize)
+{
+    std::allocator<Grenade>                                     allocator{};
+    cetl::VariableLengthArray<Grenade, std::allocator<Grenade>> subject{{Grenade{1}}, allocator};
+    ASSERT_EQ(1, subject.size());
+    ASSERT_THROW(subject.resize(2, Grenade{2}), GrenadeError);
```

Comment:
This throws before the call to resize() doesn't it?  So what is it testing?

Reply:
![image](https://github.com/OpenCyphal/CETL/assets/19180861/f4bb1523-acab-4c2f-9577-d5ed899f97ab)